datafusion/common-runtime/src/common.rs

```diff
@@ -36,8 +43,8 @@ impl<R: 'static> SpawnedTask<R> {
         T: Send + 'static,
         R: Send,
     {
-        let mut inner = JoinSet::new();
-        inner.spawn(task);
+        #[allow(clippy::disallowed_methods)]
+        let inner = tokio::task::spawn(trace_future(task));
```

Comment:
I think some rationale about why it is ok to use the disallowed methods would help. Something like:

```suggestion
        // Ok to use spawn here as SpawnedTask handles aborting/cancelling the task on Drop
        let inner = tokio::task::spawn(trace_future(task));
```